libtransmission/handshake.cc

```diff
@@ -637,6 +637,8 @@ void tr_handshake::on_error(tr_peerIo* io, tr_error const& error, void* vhandsha
         handshake->done(false);
     };
 
+    handshake->maybe_recycle_dh();
```

Comment:
#### Note to reviewers

This line might seem sketchy because the new `tr_handshake::dh_` value is going to be used as-is, and it wasn't initialised from the mediator. However, it will make complete sense when combined with #6949.